jwst/assign_wcs/nirspec.py

```diff
@@ -516,10 +527,30 @@ def detector_to_gwa(reference_files, detector, disperser):
                disperser['theta_z'], disperser['tilt_y']]
     rotation = Rotation3DToGWA(angles, axes_order="xyzy", name='rotaton')
     u2dircos = Unitless2DirCos(name='unitless2directional_cosines')
-    model = (fpa | camera | u2dircos | rotation)
+    model = (models.Shift(-1) & models.Shift(-1) | fpa | camera | u2dircos | rotation)
     return model
 
 
+def dms_to_sca(input_model):
+    """
+    Transforms from SCA to DMS coordinates.
+    """
+    detector = input_model.meta.instrument.detector
+    xstart = input_model.meta.subarray.xstart
+    ystart = input_model.meta.subarray.ystart
+    if xstart is None:
+        xstart = 1
+    if ystart is None:
+        ystart = 1
+    # The SCA coordinates are in full frame
+    subarray2full = models.Shift(xstart) & models.Shift(ystart)
+    if detector == 'NRS2':
+        model = models.Shift(-2048) & models.Shift(-2048) | models.Scale(-1) & models.Scale(-1)
+    elif detector == 'NRS1':
+        model = models.Identity(2)
```

Comment:
I see the inversion of the axes for NRS2, but it's not obvious to me whether the swapping of the x/y axes is taken into account for both detectors. To go from SCA to DMS, NRS1 is a simple x/y swap, while NRS2 is an x/y swap plus 180 deg rotation (equivalent to inverting both axes).